streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -661,6 +644,11 @@ private void ensureCopartitioning(Set<String> copartitionGroup,
                     Arrays.sort(topics);
                     throw new TopologyBuilderException(String.format("stream-thread [%s] Topics not co-partitioned: [%s]", streamThread.getName(), Utils.mkString(Arrays.asList(topics), ",")));
                 }
+            } else {
+                if (allRepartitionTopicsNumPartitions.get(topic).numPartitions == NOT_AVAILABLE) {
```

Comment:
Yes. If we have an unknown topic (c.f. the test `"unknownTopic"`) that needs repartitioning, we don't know the number of partitions for the (to be created) internal repartitioning topic. And than this case hits.